# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/style/FontFaceImpl.h
# Commit: 5e6bf07c9c27
# Full Hash: 5e6bf07c9c2783dba94107fc4b7ddf7ae466b9b6
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2022-06-29 03:44:30
# Regressor Bug: 1771493
# File Overlap Count: 1
# Description:
#   Bug 1771493 - Part 3. Abstract away buffer extraction from FontFaceImpl::InitializeSource. r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D147818
# ==============================================================================

diff -r fbfb5b7dfb30 -r 5e6bf07c9c27 layout/style/FontFaceImpl.h
--- a/layout/style/FontFaceImpl.h	Tue Jun 28 14:27:21 2022 +0000
+++ b/layout/style/FontFaceImpl.h	Tue Jun 28 14:27:21 2022 +0000
@@ -133,10 +133,10 @@
   bool HasFontData() const;
 
   /**
-   * Creates a gfxFontFaceBufferSource to represent the font data
+   * Takes the gfxFontFaceBufferSource to represent the font data
    * in this object.
    */
-  already_AddRefed<gfxFontFaceBufferSource> CreateBufferSource();
+  already_AddRefed<gfxFontFaceBufferSource> TakeBufferSource();
 
   /**
    * Gets a pointer to and the length of the font data stored in the
@@ -184,7 +184,8 @@
 
   FontFaceImpl(FontFace* aOwner, FontFaceSetImpl* aFontFaceSet);
 
-  void InitializeSource(const UTF8StringOrArrayBufferOrArrayBufferView&);
+  void InitializeSourceURL(const nsACString& aURL);
+  void InitializeSourceBuffer(uint8_t* aBuffer, uint32_t aLength);
 
   /**
    * Sets all of the descriptor values in mDescriptors using values passed
@@ -256,8 +257,7 @@
 
   // If the FontFace was constructed with an ArrayBuffer(View), this is a
   // copy of the data from it.
-  uint8_t* mSourceBuffer;
-  uint32_t mSourceBufferLength;
+  RefPtr<FontFaceBufferSource> mBufferSource;
 
   // The values corresponding to the font face descriptors, if we are not
   // a rule backed FontFace object.  For rule backed objects, we use